### PR TITLE
friends checkbox speedup

### PR DESCRIPTION
**Pull request by davidkelly**
We now just select/deselect immediately, rather than wait for backend to update it.  Will still be updated when the call completes, but now feels much more responsive.

### CODE REVIEW

File: interface/resources/qml/hifi/Pal.qml [hunk 1]
```diff
@@ -840,11 +840,7 @@ Rectangle {
                         pal.sendToScript({method: newValue ? 'addFriend' : 'removeFriend', params: model.userName});
 
                         UserActivityLogger["palAction"](newValue ? styleData.role : "un-" + styleData.role, model.sessionId);
-
-                        // http://doc.qt.io/qt-5/qtqml-syntax-propertybinding.html#creating-property-bindings-from-javascript
-                        // I'm using an explicit binding here because clicking a checkbox breaks the implicit binding as set by
-                        // "checked:" statement above.
-                        checked = Qt.binding(function() { return (model["connection"] === "friend" ? true : false)});
+                        checked = newValue;
```
Comment (howard-stearns) [inline interface/resources/qml/hifi/Pal.qml]: do we need this line? doesn't it work without it?

**Comment by davidkelly** [comment]
Test Plan

Make a connection
Check the friend box
Uncheck it
Repeat
Should be more responsive than what is in master

**Comment by hifi-gustavo** [comment]
Build console output for commit: 726395ac3eaef6983e2e76aea8ae3d5b5a388fe8 available at: https://gist.github.com/fd8450b415951d49e8c80e9e0a970f1c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10150-726395a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10150-726395a.dmg

File: interface/resources/qml/hifi/Pal.qml (hunk 1 above)
Comment (davidkelly) [inline interface/resources/qml/hifi/Pal.qml]: true - no need.  pushed fix

Comment (howard-stearns) [inline interface/resources/qml/hifi/Pal.qml]: I'm confused. It looks like line 843 still says checked = newValue;

Comment (howard-stearns) [inline interface/resources/qml/hifi/Pal.qml]: and I don't see a new checkin after 7263...

**Comment by davidkelly** [comment]
hmm weird -- thought I pushed but I didn't

**Comment by hifi-gustavo** [comment]
Build console output for commit: d64a803451ceb11db8106ae9f7874450aad12138 available at: https://gist.github.com/fd8450b415951d49e8c80e9e0a970f1c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10150-d64a803.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10150-d64a803.dmg

**Pull request merged**